### PR TITLE
Configure CircleCI

### PR DESCRIPTION
**Pull request by PaNaVTEC**

### CODE REVIEW

**Comment by neomatrix369** [comment]
Ping me when everything works and ready to merge, we might want to add some `echo`-es as we discussed earlier!

File: circle.yml [hunk 1]
```diff
@@ -29,5 +26,5 @@ deployment:
   pullrequest:
     branch: /^.*/
     commands:
-      - _utility/deploy_prb.sh $AWS_REGION $WEBSITE_INDEX $CIRCLE_BRANCH
+      - _utility/deploy_prb.sh eu-west-1 index.html "site-$CIRCLE_BRANCH"
```
Comment (neomatrix369) [inline circle.yml]: good call @PaNaVTEC, since its used only once

**Comment by PaNaVTEC** [comment]
test

**Comment by cyplo** [comment]
Is it possible to edit existing comment instead of creating new one ?
If so - I wonder what would be better.
Same comment = less spam, but easy to miss redeploys ?
Different comments=more obvious, more spammy ?

maybe we need other people to chime in ?

File: _utility/deploy_prb.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+BUCKET_NAME=$(_utility/normalize_bucket_name.sh $3)
+
+cd _site/
+
+bucketDoesNotExist() {
+  number_of_buckets_with_name=$(aws s3 ls | awk '{print $3}' | grep -c "^$1$") 
+  echo $(expr $number_of_buckets_with_name = "1") == 0 
+}
+
+BUCKET_NOT_EXISTS=$(bucketDoesNotExist $BUCKET_NAME)
```
Comment (neomatrix369) [inline _utility/deploy_prb.sh]: I would make this block into a separate function

File: _utility/deploy_prb.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+if [ $BUCKET_NOT_EXISTS ]; then
+  echo "No previous bucket with name $BUCKET_NAME found, creating a new one"
+  aws s3 mb s3://$BUCKET_NAME --region $AWS_REGION
+  aws s3 website s3://$BUCKET_NAME --index-document $WEBSITE_INDEX_DOCUMENT	
+fi
+
+echo "Synchronizing bucket: $BUCKET_NAME"
```
Comment (neomatrix369) [inline _utility/deploy_prb.sh]: Same for this block

File: _utility/send_gh_comment.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+REPO_OWNER=$1
+REPO_NAME=$2
+PR_NUMBER=$3
+AUTH_TOKEN=$4
+BUCKET_NAME=$(_utility/normalize_bucket_name.sh $5)
+
+COMMENT="Deployed: $BUCKET_NAME"
+LAST_COMMENT_ID=$(curl https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/issues/$PR_NUMBER/comments?access_token=$AUTH_TOKEN -X GET | jq '[.[] | select(.user.login=="CoduranceBot")][0].id')
+
+if [ "$LAST_COMMENT_ID" != "null" ]; then
```
Comment (neomatrix369) [inline _utility/send_gh_comment.sh]: 👍 good one, better than inside the build status thing

File: _utility/deploy_prb.sh (hunk 2 above)
Comment (neomatrix369) [inline _utility/deploy_prb.sh]: minor: variable name: `BUCKET_DOES_NOT_EXIST`

File: circle.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  environment:
+    PR_NUMBER: $(echo ${CI_PULL_REQUEST##*/})
+    BUCKET_NAME: "site-$CIRCLE_BRANCH"
+    WEBSITE_INDEX_DOCUMENT: index.html
+    PREVIEW_URL: "https://${BUCKET_NAME}.s3.amazonaws.com/$WEBSITE_INDEX_DOCUMENT"
+    AWS_REGION: eu-west-1
+dependencies:
+  cache_directories:
+    - "/opt/circleci/.rvm/gems"
+    - "_site"
```
Comment (neomatrix369) [inline circle.yml]: Does this work? Or is an absolute path to `/site/` is needed?

File: circle.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  pre:
+    - sudo apt-get install jq
```
Comment (neomatrix369) [inline circle.yml]: `jq` is cool, have used in the past 👍

File: _utility/send_gh_comment.sh (hunk 4 above)
Comment (cyplo) [inline _utility/send_gh_comment.sh]: not sure intended but `[[` provide fuller evaluation than `[` does

**Pull request merged**